### PR TITLE
[MAINT] refactor tests glm

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Relates to #3570

Changes proposed in this pull request:

- refactor several complex tests in nilearn/glm/tests
- use more fixtures

### CODE REVIEW

File: nilearn/glm/tests/test_thresholding.py [hunk 1]
```diff
@@ -34,93 +52,120 @@ def test_fdr():
...
-        mask_img,
+        mask_img=img_3d_ones,
         threshold=100,
         height_control=None,
         cluster_threshold=0,
     )
     vals = get_data(th_map)
     assert np.sum(vals > 0) == 0
 
-    # test 3: excessive size threshold
```
Comment (bthirion) [inline nilearn/glm/tests/test_thresholding.py:64]: Is this one gone ?

**Review by bthirion — COMMENTED** [review]
LGTM so far.

File: nilearn/glm/tests/test_thresholding.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/glm/tests/test_thresholding.py:64]: nope moved below (line 127) in a separate test that have `height_control != None`

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**